runtime/sema/type.go

```diff
@@ -3882,6 +3898,19 @@ func (t *CompositeType) MemberMap() *StringMemberOrderedMap {
 	return t.Members
 }
 
+func (t *CompositeType) SupportedEntitlements() (set *EntitlementOrderedSet) {
+	set = orderedmap.New[EntitlementOrderedSet](t.Members.Len())
+	t.Members.Foreach(func(_ string, member *Member) {
```

Comment:
I guess this includes any default functions coming from interfaces? i.e: entitlement isn't directly used in the composite type.
Maybe good to add a test if there aren't any.

Reply:
Good call; this wasn't actually working properly. Fixed it and added a test.